src/fr/raijinscans/src/eu/kanade/tachiyomi/extension/fr/raijinscans/RaijinScans.kt

```diff
@@ -1,9 +1,205 @@
+        val url = baseUrl.toHttpUrl().newBuilder().apply {
+            if (page > 1) addPathSegment("page").addPathSegment(page.toString())
+            addQueryParameter("s", query)
+            addQueryParameter("post_type", "wp-manga")
+        }.build()
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.select("div.original.card-lg div.unit").map(::searchMangaFromElement)
+        val hasNextPage = document.selectFirst("li.page-item:not(.disabled) a[rel=next]") != null
+        return MangasPage(mangas, hasNextPage)
+    }
+
+    private fun searchMangaFromElement(element: Element): SManga = SManga.create().apply {
+        val linkElement = element.selectFirst("div.info > a")!!
+        setUrlWithoutDomain(linkElement.attr("abs:href"))
+        title = linkElement.text()
+        thumbnail_url = element.selectFirst("div.poster-image-wrapper > img")?.attr("abs:src")
+    }
+
+    // =========================== Manga Details ===========================
+    private fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        title = document.selectFirst("h1.serie-title")!!.text()
+        author = document.selectFirst("div.stat-item:has(span:contains(Auteur)) span.stat-value")?.text()
+        artist = document.selectFirst("div.stat-item:has(span:contains(Artiste)) span.stat-value")?.text()
+
+        val scriptDescription = document.select("script:containsData(content.innerHTML)")
+            .firstNotNullOfOrNull { descriptionScriptRegex.find(it.data())?.groupValues?.get(1)?.trim() }
+        description = scriptDescription ?: document.selectFirst("div.description-content")?.text()
+
+        genre = document.select("div.genre-list div.genre-link").joinToString { it.text() }
+
+        thumbnail_url = document.selectFirst("img.cover")?.attr("abs:src")
+        status = parseStatus(document.selectFirst("div.stat-item:has(span:contains(État du titre)) span.manga")?.text())
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga = mangaDetailsParse(response.asJsoup())
+
+    private fun parseStatus(status: String?): Int {
+        return when {
+            status == null -> SManga.UNKNOWN
+            status.contains("En cours", ignoreCase = true) -> SManga.ONGOING
+            status.contains("Terminé", ignoreCase = true) -> SManga.COMPLETED
+            else -> SManga.UNKNOWN
+        }
+    }
+
+    // ========================= Chapter List ==========================
+    override fun chapterListParse(response: Response): List<SChapter> {
+        return response.asJsoup().select("ul.scroll-sm li.item").map(::chapterFromElement)
+    }
+
+    private fun chapterFromElement(element: Element): SChapter = SChapter.create().apply {
+        val link = element.selectFirst("a")!!
+        setUrlWithoutDomain(link.attr("href"))
```

Comment:
To let Jsoup resolve relative URLs:
```suggestion
        setUrlWithoutDomain(link.absUrl("href"))
```
```suggestion
        setUrlWithoutDomain(link.attr("abs:href"))
```